ndonnx/_typed_array/object_dtype.py

```diff
@@ -0,0 +1,347 @@
+# Copyright (c) QuantCo 2023-2025
+# SPDX-License-Identifier: BSD-3-Clause
+"""Implementation of a "object" dtype that implements a str | None | np.nan enum."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, cast
+
+import numpy as np
+
+from .._dtypes import TY_ARRAY_BASE, DType
+from .._schema import DTypeInfoV1
+from . import onnx
+from .funcs import astyarray, where
+from .typed_array import TyArrayBase
+from .utils import safe_cast
+
+if TYPE_CHECKING:
+    from spox import Var
+    from typing_extensions import Self
+
+    from .._types import NestedSequence, OnnxShape, PyScalar
+    from .indexing import GetitemIndex, SetitemIndex
+
+
+# TODO: The name is unfortunate. We cover far less than the "object"
```

Comment:
rm or move into test suite? Better in downstream library in my opinion, as it is too specific to pandas.

Reply:
Move to the [`object-array`](https://github.com/Quantco/ndonnx/tree/object-array) branch for now.